app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/maps/MapsFragmentTest.kt

```diff
@@ -19,20 +20,32 @@ import androidx.test.uiautomator.UiDevice
 import androidx.test.uiautomator.UiSelector
 import com.github.factotum_sdp.factotum.MainActivity
 import com.github.factotum_sdp.factotum.R
+import com.google.android.gms.maps.SupportMapFragment
 import junit.framework.TestCase.assertTrue
 import org.hamcrest.CoreMatchers.allOf
 import org.hamcrest.Matchers.anything
 import org.junit.Assert.assertEquals
 import org.junit.Before
+import org.junit.FixMethodOrder
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+import java.util.*
+import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 
 @RunWith(AndroidJUnit4::class)
+//used to force first the tests about permissions
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 class MapsFragmentTest {
 
+    val device = UiDevice.getInstance(getInstrumentation())
+    val buttonTextAllow = when(Locale.getDefault().language){
```

Comment:
Isn't there another way of getting the "Only this time" for any language?
This is hardcoded for only two languages and limits the test.

Reply:
I understand but I didn't found a way to automate this. And using a translation isn't reliable (ie Deepl has translated "Only this time" -> "Mais cette fois-ci"). For now english and french seem to be the only needed languages. But I agree that this test should be adapted if more languages would be involved

Reply:
I will add it in the backlog